### ISSUE TITLE
discussion: map adapter

### ISSUE THREAD

**Issue by najeira** [syumai/go-jpostcode#23]
I implemented `map adapter` that uses `map` and `gob`.
Benchmark: https://github.com/najeira/go-jpostcode/pull/1
Map version: https://github.com/najeira/go-jpostcode/pull/2

I would like to compare it with the `badger adapter` and discuss which one is better.

## Benchmark results

Map takes about 20% slower to initialize, but uses half the memory:

```
Benchmark_newBadgerAdapter-8       	       8	 143236990 ns/op	239397165 B/op	 1994414 allocs/op
Benchmark_newMapAdapter-8          	       6	 172011243 ns/op	94680586 B/op	 1834982 allocs/op
```

Map is 1000x faster:

```
Benchmark_BadgerAdapter_Search-8   	  110256	     11474 ns/op	    9889 B/op	     132 allocs/op
Benchmark_MapAdapter_Search-8      	153411405	         7.828 ns/op	       0 B/op	       0 allocs/op
```

----

Please let me know your opinion on this.
And, please tell me the background and thoughts when you chose `badger` now.

**Comment by syumai** [comment]
@najeira 
I had tried an implementation using map in the past, but it failed to launch the application and did not work (probably due to memory pressure).
However, looking at your benchmarks, map implementation looks fine. Perhaps my implementation at the time was not efficient.

The pull request looks fine overall, so would you please submit it for this repository? Thanks.

**Comment by syumai** [comment]
#24 has been merged. Thank you @najeira! 🎉

**Comment by syumai** [comment]
This feature has been released as [v0.2.4](https://github.com/syumai/go-jpostcode/releases/tag/v0.2.4).

**Comment by najeira** [comment]
By the way, I just tried to generate a file that declares addresses in Go's map literal, but it wouldn't compile.

Maybe Go can't compile a huge file.

My PR uses gob binaries, so it worked around the problem you have faced in the past.
